sqlmesh/core/engine_adapter/spark.py

```diff
@@ -38,14 +44,53 @@ def spark(self) -> PySparkSession:
     def _use_spark_session(self) -> bool:
         return True
 
-    def _ensure_pyspark_df(self, generic_df: DF) -> PySparkDataFrame:
+    @classmethod
+    def convert_columns_to_types_to_pyspark_schema(
+        cls, columns_to_types: t.Dict[str, exp.DataType]
+    ) -> t.Optional[spark_types.StructType]:
+        from pyspark.sql import types as spark_types
+
+        mapping = {
```

Comment:
make this a constant

Reply:
Ended up not doing this since it requires pyspark types and we don't want to require importing that so I left it where it is.